src/client/mod.rs

```diff
@@ -82,7 +81,7 @@ impl Client {
         options.validate()?;
 
         let inner = Arc::new(ClientInner {
-            topology: RUNTIME.block_on(Topology::new(options.clone()))?,
+            topology: Topology::new(options.clone())?,
```

Comment:
`Topology::new` didn't need to be async, since the only blocking behavior it had was waiting on a lock that it created itself. I refactored some stuff to make acquiring that lock not necessary, and this subsequently allowed me to make `Client::with_options` not async.

Reply:
Good work on this! I'm honestly not sure this will ever need to be called in a non-async context, but the code is definitely cleaner regardless.